### ISSUE TITLE
Provisioning CF fails with Error 100: no implicit conversion of nil into String

### ISSUE THREAD

**Issue by dzlab** [cloudfoundry/bosh-lite#233]
I'm trying to setup cloud foundry on my local machine (Windows 7), when I try to provision CF (after installing bosh lite and spiff) lot if thing happens then it fails. Here is the output of the provisioning script:
```
$ cd bosh-lite
$ ./bin/provision_cf
Target already set to `Bosh Lite Director'

Verifying stemcell...
File exists and readable                                     OK
Verifying tarball...
Read tarball                                                 OK
Manifest exists                                              OK
Stemcell image file                                          OK
Stemcell properties                                          OK

Stemcell info
-------------
Name:    bosh-warden-boshlite-ubuntu-trusty-go_agent
Version: 389

Checking if stemcell already exists...
Stemcell `bosh-warden-boshlite-ubuntu-trusty-go_agent/389' already exists. Skipping upload.

===> Uncommitted submodules changes will be clobbered <===


===> Unversioned changes will be clobbered <===

Already up-to-date.
Synchronizing submodule url for 'packages/ruby'
Synchronizing submodule url for 'packages/ruby-2.1.4'
Synchronizing submodule url for 'shared'
Synchronizing submodule url for 'src/acceptance-tests'
Synchronizing submodule url for 'src/cloud_controller_ng'
Synchronizing submodule url for 'src/collector'
Synchronizing submodule url for 'src/dea_next'
Synchronizing submodule url for 'src/etcd'
Synchronizing submodule url for 'src/etcd-metrics-server'
Synchronizing submodule url for 'src/gnatsd'
Synchronizing submodule url for 'src/gorouter'
Synchronizing submodule url for 'src/hm9000'
Synchronizing submodule url for 'src/loggregator'
Synchronizing submodule url for 'src/login'
Synchronizing submodule url for 'src/smoke-tests'
Synchronizing submodule url for 'src/uaa'
Synchronizing submodule url for 'src/warden'
Entering 'packages/ruby'
Entering 'packages/ruby-2.1.4'
Entering 'shared'
Entering 'src/acceptance-tests'
Entering 'src/cloud_controller_ng'
Synchronizing submodule url for 'vendor/errors'
Entering 'src/cloud_controller_ng/vendor/errors'
Entering 'src/collector'
Entering 'src/dea_next'
Synchronizing submodule url for 'go/src/github.com/cloudfoundry/gosteno'
Synchronizing submodule url for 'go/src/github.com/howeyc/fsnotify'
Entering 'src/dea_next/go/src/github.com/cloudfoundry/gosteno'
Entering 'src/dea_next/go/src/github.com/howeyc/fsnotify'
Entering 'src/etcd'
Entering 'src/etcd-metrics-server'
Entering 'src/gnatsd'
Entering 'src/gorouter'
Entering 'src/hm9000'
Synchronizing submodule url for 'src/github.com/apcera/nats'
Synchronizing submodule url for 'src/github.com/bmizerany/pat'
Synchronizing submodule url for 'src/github.com/ccding/go-config-reader'
Synchronizing submodule url for 'src/github.com/ccding/go-logging'
Synchronizing submodule url for 'src/github.com/cloudfoundry-incubator/natbeat'
Synchronizing submodule url for 'src/github.com/cloudfoundry/dropsonde'
Synchronizing submodule url for 'src/github.com/cloudfoundry/go_cfmessagebus'
Synchronizing submodule url for 'src/github.com/cloudfoundry/gosteno'
Synchronizing submodule url for 'src/github.com/cloudfoundry/gunk'
Synchronizing submodule url for 'src/github.com/cloudfoundry/hm9000'
Synchronizing submodule url for 'src/github.com/cloudfoundry/loggregatorlib'
Synchronizing submodule url for 'src/github.com/cloudfoundry/storeadapter'
Synchronizing submodule url for 'src/github.com/cloudfoundry/yagnats'
Synchronizing submodule url for 'src/github.com/codegangsta/cli'
Synchronizing submodule url for 'src/github.com/coreos/etcd'
Synchronizing submodule url for 'src/github.com/coreos/go-etcd'
Synchronizing submodule url for 'src/github.com/coreos/go-log'
Synchronizing submodule url for 'src/github.com/coreos/go-systemd'
Synchronizing submodule url for 'src/github.com/goji/httpauth'
Synchronizing submodule url for 'src/github.com/gorilla/websocket'
Synchronizing submodule url for 'src/github.com/nu7hatch/gouuid'
Synchronizing submodule url for 'src/github.com/onsi/ginkgo'
Synchronizing submodule url for 'src/github.com/onsi/gomega'
Synchronizing submodule url for 'src/github.com/pivotal-golang/lager'
Synchronizing submodule url for 'src/github.com/pivotal-golang/localip'
Synchronizing submodule url for 'src/github.com/samuel/go-zookeeper'
Synchronizing submodule url for 'src/github.com/stretchr/testify'
Synchronizing submodule url for 'src/github.com/tedsuo/ifrit'
Synchronizing submodule url for 'src/github.com/tedsuo/rata'
Synchronizing submodule url for 'src/tux21b.org/v1/gocql'
Entering 'src/hm9000/src/github.com/apcera/nats'
Entering 'src/hm9000/src/github.com/bmizerany/pat'
Entering 'src/hm9000/src/github.com/ccding/go-config-reader'
Entering 'src/hm9000/src/github.com/ccding/go-logging'
Entering 'src/hm9000/src/github.com/cloudfoundry-incubator/natbeat'
Entering 'src/hm9000/src/github.com/cloudfoundry/dropsonde'
Synchronizing submodule url for 'dropsonde-protocol'
Entering 'src/hm9000/src/github.com/cloudfoundry/dropsonde/dropsonde-protocol'
Enter+ git submodule update --init --recursive --force
+ git clean -ffd
+ set +x
+ cd /e/Users/dzlab/vms/bosh-lite
+ export CF_RELEASE_DIR=/e/Users/dzlab/vms/cf-release
+ CF_RELEASE_DIR=/e/Users/dzlab/vms/cf-release
+ ./bin/make_manifest_spiff
Can't do inplace edit on ./bin/../tmp/boshlite-10196.tmp/cf-stub-with-uuid.yml: Permission denied.
2015/01/27 15:18:48 error reading stub: open ./bin/../tmp/boshlite-10196.tmp/cf-stub-with-uuid.yml: The system cannot find the file specified.
Incorrect YAML structure in `e:/Users/dzlab/vms/bosh-lite/manifests/cf-manifest.yml': expected Hash at the root+ deploy_release
++ find /e/Users/dzlab/vms/cf-release/releases -regex '.*cf-[0-9]*.yml'
++ sort
++ tail -n 1
+ MOST_RECENT_CF_RELEASE=/e/Users/dzlab/vms/cf-release/releases/cf-196.yml
+ bosh -n -u admin -p admin upload release --skip-if-exists /e/Users/dzlab/vms/cf-release/releases/cf-196.yml
'src/hm9000/src/github.com/onsi/gomega'
Entering 'src/hm9000/src/github.com/pivotal-golang/lager'
Entering 'src/hm9000/src/github.com/pivotal-golang/localip'
Entering 'src/hm9000/src/github.com/samuel/go-zookeeper'
Entering 'src/hm9000/src/github.com/stretchr/testify'
Entering 'src/hm9000/src/github.com/tedsuo/ifrit'
Entering 'src/hm9000/src/github.com/tedsuo/rata'
Entering 'src/hm9000/src/tux21b.org/v1/gocql'
Entering 'src/loggregator'
Synchronizing submodule url for 'src/code.google.com/p/gogoprotobuf'
Synchronizing submodule url for 'src/github.com/apcera/gnatsd'
Synchronizing submodule url for 'src/github.com/apcera/nats'
Synchronizing submodule url for 'src/github.com/cloudfoundry/dropsonde'
Synchronizing submodule url for 'src/github.com/cloudfoundry/gosteno'
Synchronizing submodule url for 'src/github.com/cloudfoundry/gunk'
Synchronizing submodule url for 'src/github.com/cloudfoundry/loggregator_consumer'
Synchronizing submodule url for 'src/github.com/cloudfoundry/loggregatorlib'
Synchronizing submodule url for 'src/github.com/cloudfoundry/noaa'
Synchronizing submodule url for 'src/github.com/cloudfoundry/storeadapter'
Synchronizing submodule url for 'src/github.com/cloudfoundry/yagnats'
Synchronizing submodule url for 'src/github.com/coreos/etcd'
Synchronizing submodule url for 'src/github.com/coreos/go-etcd'
Synchronizing submodule url for 'src/github.com/davecgh/go-spew'
Synchronizing submodule url for 'src/github.com/gorilla/websocket'
Synchronizing submodule url for 'src/github.com/howeyc/fsnotify'
Synchronizing submodule url for 'src/github.com/nu7hatch/gouuid'
Synchronizing submodule url for 'src/github.com/onsi/ginkgo'
Synchronizing submodule url for 'src/github.com/onsi/gomega'
Synchronizing submodule url for 'src/github.com/stretchr/testify'
Entering 'src/loggregator/src/code.google.com/p/gogoprotobuf'
Entering 'src/loggregator/src/github.com/apcera/gnatsd'
Entering 'src/loggregator/src/github.com/apcera/nats'
Entering 'src/loggregator/src/github.com/cloudfoundry/dropsonde'
Synchronizing submodule url for 'dropsonde-protocol'
Entering 'src/loggregator/src/github.com/cloudfoundry/dropsonde/dropsonde-protocol'
Entering 'src/loggregator/src/github.com/cloudfoundry/gosteno'
Entering 'src/loggregator/src/github.com/cloudfoundry/gunk'
Entering 'src/loggregator/src/github.com/cloudfoundry/loggregator_consumer'
Entering 'src/loggregator/src/github.com/cloudfoundry/loggregatorlib'
Entering 'src/loggregator/src/github.com/cloudfoundry/noaa'
Entering 'src/loggregator/src/github.com/cloudfoundry/storeadapter'
Entering 'src/loggregator/src/github.com/cloudfoundry/yagnats'
Entering 'src/loggregator/src/github.com/coreos/etcd'
Entering 'src/loggregator/src/github.com/coreos/go-etcd'
Entering 'src/loggregator/src/github.com/davecgh/go-spew'
Entering 'src/loggregator/src/github.com/gorilla/websocket'
Entering 'src/loggregator/src/github.com/howeyc/fsnotify'
Entering 'src/loggregator/src/github.com/nu7hatch/gouuid'
Entering 'src/loggregator/src/github.com/onsi/ginkgo'
Entering 'src/loggregator/src/github.com/onsi/gomega'
Entering 'src/loggregator/src/github.com/stretchr/testify'
Entering 'src/login'
Synchronizing submodule url for 'uaa'
Entering 'src/login/uaa'
Entering 'src/smoke-tests'
Entering 'src/uaa'
Entering 'src/warden'
Submodule path 'packages/ruby': checked out '01b9b845268ba49041068bb877452ee3e242682f'
Submodule path 'packages/ruby-2.1.4': checked out '5f852875676b174ed17d1e52f7ccf960ce7b284f'
Submodule path 'shared': checked out 'e6eac0d0b8ced9a1c2d4a4ff82f19b104ea1f042'
Submodule path 'src/acceptance-tests': checked out 'ccf79c037d03987a5521cbac998ba3379398ba26'
Submodule path 'src/cloud_controller_ng': checked out 'b0e4288d4f368f9ae17378d801a482f6b829ca80'
Submodule path 'src/cloud_controller_ng/vendor/errors': checked out 'b4aa6f3996e3d681ce3010c699511e5d73877ed0'
Submodule path 'src/collector': checked out 'ffea2b49176878d0d1284c3d6d0ccddc1053738d'
Submodule path 'src/dea_next': checked out 'e03f89dcf6a75764e130be41c5ca036c4a2377e4'
Submodule path 'src/dea_next/go/src/github.com/cloudfoundry/gosteno': checked out 'c6379cb7ef097850eec4dc61e1730bbb99a2a2a8'
Submodule path 'src/dea_next/go/src/github.com/howeyc/fsnotify': checked out '08040c5a90632bd721465eb8ad74a8e61bd7bf95'
Submodule path 'src/etcd': checked out '49e0dff2b8529801beb09164729caf96a5b93ef0'
Submodule path 'src/etcd-metrics-server': checked out '90c444c7f93cacb998e45c46f1e06ecf4c8eb9c4'
Submodule path 'src/gnatsd': checked out '7fb53586108b89a68d3db9ad2f7cf84a6842194c'
Submodule path 'src/gorouter': checked out '13e2cce87c81edb3bad1a317583a680e0c44101a'
Submodule path 'src/hm9000': checked out '596ca8cd97a6a7391bd927ede4c2c89c6a775fe1'
Submodule path 'src/hm9000/src/github.com/apcera/nats': checked out '4ca550d13b9b6d03b2dded15287a1eae2ae27fa8'
Submodule path 'src/hm9000/src/github.com/bmizerany/pat': checked out 'b8a35001b773c267eb260a691f4e5499a3531600'
Submodule path 'src/hm9000/src/github.com/ccding/go-config-reader': checked out '8b6c2b50197f20da3b1c5944c274c173634dc056'
Submodule path 'src/hm9000/src/github.com/ccding/go-logging': checked out 'c0585d19814ac8e8a37b9253bf25cc6152ba7a6b'
Submodule path 'src/hm9000/src/github.com/cloudfoundry-incubator/natbeat': checked out '3eb0da93534108b6aab163d53093a1b3d79d6bbc'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/dropsonde': checked out 'e4e7f2747fc5f86e9ff9ce5bf6d3b1ed954cc1a6'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/dropsonde/dropsonde-protocol': checked out '959cb0e2b48a9f5fb6c0db0a8a930cc9b10e3751'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/go_cfmessagebus': checked out '383a83e25e4df446dcc3524a42b43f5a6b73a355'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/gosteno': checked out 'b27c580bf20520f61cb21f82365c1ef3b8e601e9'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/gunk': checked out 'de79b69f590594db2b9771018784c55251d5e7e7'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/hm9000': checked out '9b19a802487cd03810b0eefacc97587aa3105395'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/loggregatorlib': checked out '8dbdbfdc025b4b5b044265a1ec6e7480391eb4ee'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/storeadapter': checked out '5dae4c33df48f65e22e7f4c4326f975f4b887636'
Submodule path 'src/hm9000/src/github.com/cloudfoundry/yagnats': checked out '1ac54a8c5bcd394f3904dbad5f4d10fea5241069'
Submodule path 'src/hm9000/src/github.com/codegangsta/cli': checked out '3c97f95b6a61fe9bf6d56f33418599d8a7265fca'
Submodule path 'src/hm9000/src/github.com/coreos/etcd': checked out '9970141f76241c909977af7bafe7b6f2c4923de8'
Submodule path 'src/hm9000/src/github.com/coreos/go-etcd': checked out '0f26d821ecf2f79984e03606b1de9d775b175092'
Submodule path 'src/hm9000/src/github.com/coreos/go-log': checked out '70d039bee4b0e389e5be560491d8291708506f59'
Submodule path 'src/hm9000/src/github.com/coreos/go-systemd': checked out 'b6bbed6dfecfe5cae49a38330b8ddb730cbd741b'
Submodule path 'src/hm9000/src/github.com/goji/httpauth': checked out '5279c8de0d24f35f88205340317b30b31084892c'
Submodule path 'src/hm9000/src/github.com/gorilla/websocket': checked out 'a6f041ac33b84488bb44137d6866cd3c8706d792'
Submodule path 'src/hm9000/src/github.com/nu7hatch/gouuid': checked out '87bcc4729f2c5a08d2513ad10684c6bbd256380f'
Submodule path 'src/hm9000/src/github.com/onsi/ginkgo': checked out '4e965bd136569a9db429768a24b5789b473c8e31'
Submodule path 'src/hm9000/src/github.com/onsi/gomega': checked out '91ddcd2375dd05fecea5eabf7a8f86332de756f4'
Submodule path 'src/hm9000/src/github.com/pivotal-golang/lager': checked out 'a106931abed5afe3043eae8851be8b33a4bdfa3f'
Submodule path 'src/hm9000/src/github.com/pivotal-golang/localip': checked out 'ca5f12419a47fe0c8547eea32f9498eb6e9fe817'
Submodule path 'src/hm9000/src/github.com/samuel/go-zookeeper': checked out '897888e14f7bd6c1c9a70e2973d0f5fbdb28bde0'
Submodule path 'src/hm9000/src/github.com/stretchr/testify': checked out 'de7fcff264cd05cc0c90c509ea789a436a0dd206'
Submodule path 'src/hm9000/src/github.com/tedsuo/ifrit': checked out '89cad2291fb2164d0910192dfc9c91046a8d02b6'
Submodule path 'src/hm9000/src/github.com/tedsuo/rata': checked out '9946bb39c19484e199efb807da145063323e6502'
Submodule path 'src/hm9000/src/tux21b.org/v1/gocql': checked out '44eda643c1ae69e866e491b1c935c5b22e42350e'
Submodule path 'src/loggregator': checked out 'eadf1a702cfa1174b7f89eb40f168aa6155b801a'
Submodule path 'src/loggregator/src/code.google.com/p/gogoprotobuf': checked out '6c980277330804e94257ac7ef70a3adbe1641059'
Submodule path 'src/loggregator/src/github.com/apcera/gnatsd': checked out '7d820b762fbcc58264b7d688bd869e5949e8d175'
Submodule path 'src/loggregator/src/github.com/apcera/nats': checked out 'a85ef48dcb52a020cd3f301981ff9e89b8aaf369'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/dropsonde': checked out 'b05fce8468d0aefa3a90f5ce731696e7897f1654'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/dropsonde/dropsonde-protocol': checked out '8d096c34eda1827a35f064496488da6e247a9b08'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/gosteno': checked out '5c1406e9ea0ab919f92e1b194169565def41147b'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/gunk': checked out 'bf8c7bb0c1cd0c4ae29bd952b49d6bee8ce27e92'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/loggregator_consumer': checked out 'c398c00372d652a8ef6aae475cfcddc32db1203c'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/loggregatorlib': checked out '1c010a67576de3da7ae9ea3583575b55d3ab7b1a'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/noaa': checked out 'da595891ae163425fd8f7d23b2caa3c69c0edf2e'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/storeadapter': checked out '652efe1a1fcf3fdb5b08d7474594eb250be6009c'
Submodule path 'src/loggregator/src/github.com/cloudfoundry/yagnats': checked out 'db9f0efeb22cc77857bc628fc35fc89a806eb7a8'
Submodule path 'src/loggregator/src/github.com/coreos/etcd': checked out '49e0dff2b8529801beb09164729caf96a5b93ef0'
Submodule path 'src/loggregator/src/github.com/coreos/go-etcd': checked out 'e10c58ee110f54c2f385ac99764e8a7ca4cb13df'
Submodule path 'src/loggregator/src/github.com/davecgh/go-spew': checked out '3fdaf5cea8d23107b993d363e98992fa529dd713'
Submodule path 'src/loggregator/src/github.com/gorilla/websocket': checked out '1e6e1281b05fe5eaaf3300bdedb8e75880b9c6fd'
Submodule path 'src/loggregator/src/github.com/howeyc/fsnotify': checked out '441bbc86b167f3c1f4786afae9931403b99fdacf'
Submodule path 'src/loggregator/src/github.com/nu7hatch/gouuid': checked out '179d4d0c4d8d407a32af483c2354df1d2c91e6c3'
Submodule path 'src/loggregator/src/github.com/onsi/ginkgo': checked out '6bd31378f9b5c099a60ab96b89f8fa9b48678d44'
Submodule path 'src/loggregator/src/github.com/onsi/gomega': checked out '4699d5a664396173c7d9fcef9e88be297c7bae13'
Submodule path 'src/loggregator/src/github.com/stretchr/testify': checked out '9cc77fa25329013ce07362c7742952ff887361f2'
Submodule path 'src/login': checked out '982ffdc79a156acca71d0c32e2942fe2157dd77e'
Submodule path 'src/login/uaa': checked out '9156eed2dcc13d8aa1198523d5ce3876e3fe61d7'
Submodule path 'src/smoke-tests': checked out '467a8c41347d67631ea0732693f24efc99fa3d69'
Submodule path 'src/uaa': checked out '9156eed2dcc13d8aa1198523d5ce3876e3fe61d7'
Submodule path 'src/warden': checked out 'dbde59424a6cdbeae3a265c1c8ce5a37f8f4ebae'
CF Release has been updated
35b3e645-7b07-435a-bfab-69f718e0e8ef

Config
             c:/Users/dzlab/.bosh_config

Director
  Name       Bosh Lite Director
  URL        https://192.168.50.4:25555
  Version    1.2776.0 (00000000)
  User       admin
  UUID       35b3e645-7b07-435a-bfab-69f718e0e8ef
  CPI        vsphere
  dns        disabled
  compiled_package_cache enabled (provider: local)
  snapshots  enabled 

Deployment
  not set

Using remote release `e:/Users/dzlab/vms/cf-release/releases/cf-196.yml'

Director task 9
  Started downloading remote release > Downloading remote release. Failed: no implicit conversion of nil into String (00:00:00)

Error 100: no implicit conversion of nil into String

Task 9 error

For a more detailed error report, run: bosh task 9 --debug
```

**Comment by cppforlife** [comment]
We do not officially support bosh CLI for windows. It seems that it's failing to upload a release because it thinkgs that e: drive is a remote destination. I would recommend using a Mac/Linux environment - you can even `vagrant ssh` into the bosh-lite VM and do it from there.

**Comment by dkoper** [comment]
@dzlab, I have fixed a few bosh CLI on Windows issues in the past. https://github.com/cloudfoundry/bosh/pull/630 refers to an issue with (cygwin?) tar on Windows that is still unresolved. This issue can occur in a number of paths where ```tar``` is invoked, including the one in your issue.
I gave up on fixing them when I discovered I could use the bosh CLI installed within the bosh-lite Ubuntu VM instead.
